python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -653,9 +683,8 @@ def _init_db(self):
         # as UTF-8 (byte string) or unicode. And in the latter case, the returned data
         # will always be unicode.
         connection.text_factory = str
-        c = connection.cursor()
 
-        try:
+        with contextlib.closing(connection.cursor()) as c:
```

Comment:
I don't know why the cursor doesn't have `__enter__` `__exit__` implemented on them it seems such an obvious feature to add.
maybe I should submit a PR to cpython :p